Remotion/Web/Core/UI/Controls/WebTabCollection.cs

```diff
@@ -148,23 +148,13 @@ protected void DeselectTab (WebTab tab, int index)
 
       if (_tabStrip != null && tab.IsSelected)
       {
-        bool isLastTab = index + 1 == InnerList.Count;
-        if (isLastTab)
-        {
-          if (InnerList.Count > 1)
-          {
-            WebTab penultimateTab = (WebTab)InnerList[index - 1]!;
-            _tabStrip.SetSelectedTabInternal(penultimateTab);
-          }
-          else
-            _tabStrip.SetSelectedTabInternal(null);
-        }
-        else
-        {
-          WebTab nextTab = (WebTab)InnerList[index + 1]!;
-          _tabStrip.SetSelectedTabInternal(nextTab);
-        }
+        var webTabs = InnerList.Cast<WebTab>().ToList();
+        var nextTab = webTabs.Skip(index + 1).FirstOrDefault(IsSelectable) ?? webTabs.Take(index).LastOrDefault(IsSelectable);
```

Comment:
readability and allocation prevention (the ToList isn't needed, we can just to a cast twice):
```suggestion
        var tabsBeforeSelectedTab = InnerList.Cast<WebTab>().Take(index);
        var tabsAfterSelectedTab = InnerList.Cast<WebTab>().Skip(index + 1);
        var nextTab = tabsAfterSelectedTab.FirstOrDefault(IsSelectable) ?? tabsBeforeSelectedTab .LastOrDefault(IsSelectable);
```

If we didn't incur the allocation cost for the array, we could also do C# Range syntax.